### ISSUE TITLE
White screen for AH on trying to update schedule dates (when hitting save).  When returning to page results are not saved.

### ISSUE THREAD

**Issue by micheal-w-wells** [bcgov/range-web#339]
@LisaMoore1 this is Todd's issue.

**Comment by LisaMoore1** [comment]
@micheal-w-wells. Is this one you will be working up on in the next week? If so can you move it to Sprint Backlog (if you aren’t working on it today) or In Progress if it is something you are already picking at? Thx.

**Comment by LisaMoore1** [comment]
@calebissharp @micheal-w-wells Still crashing. Let me know if I should be testing something different?

**Comment by calebissharp** [comment]
@LisaMoore1 Change hasn't been merged yet.

**Comment by LisaMoore1** [comment]
@calebissharp OK. Let me know when you want me to test that...and if I can log into test to test I will.

**Comment by LisaMoore1** [comment]
@calebissharp Used RAN099955. 
Copy schedule — update dates = no errors
Update date in a copy scheduled = white screen
Copy schedule (with rows) — save — error message — save again — error message — back to work queue page — back into RUP — there is a schedule entry for that year with no rows for each time I hit save (ex. If I hit save three times while in process of making a 2022 schedule there will three 2022 rows all with no rows) See text in “schedule description” in image below. 
Ability to delete whole schedule for selected year unpredictable
Ability to delete row in schedule unpredictable (looks like maybe only last row...but sometimes never...)

**Intermittent Errors that I can’t predictably reproduce:**
Copy schedule — save draft = error (top right “Error saving draft”) —> I think this is if I try to save a schedule that actually has rows. If I copy one with no rows I seem to be able to save
RAN099952 is in “Make Changes to RUP” but the “Add schedule” button is disabled
RAN099951 is also in “Make Change to RUP” but the “Add schedule” button is active

![2A88DE86-09A7-4BBB-820C-FAAF8858831C.jpeg](https://images.zenhubusercontent.com/5d937161c8e4d80001227881/9321af5e-2e76-4772-b849-425c3f7fe50f)

**Comment by LisaMoore1** [comment]
@calebissharp — Other observation: there is something going on where if I copy a schedule things behave differently than is I manually add a schedule.

**Comment by calebissharp** [comment]
@LisaMoore1 Could you create a separate issue for the "Add schedule" button problem, please? I don't think it's related to the other schedule bugs.

After investigating a bit, all these issues seem to stem from the schedule copy behaviour. I've got a fix in #356 that I think will cover all these cases. Were there any bugs you saw that definitely weren't related to copying?

**Comment by LisaMoore1** [comment]
@calebissharp Just inability for an AH to delete a schedule that was created by a staff person in staff draft and then submitted to them.

**Comment by LisaMoore1** [comment]
@calebissharp Also noticed that for those with multiple years of schedule same year (i.e. the ones I used for testing this card) the "Submit" option doesn't work -- it refocuses the screen back to the schedule section. I thought maybe this was because some of the schedules might not be valid (ex. not having any rows) but I updated RAN099952 so that all of the schedules are valid and it still refocuses to this section and doesn't trigger the submit workflow.  I'd like to wait until after I test your recent fix to see if the problem persists.  It is hard for me to really test until the option to delete a schedule becomes valid again so I can get rid of the duplicates.

**Comment by LisaMoore1** [comment]
@calebissharp I've got the unable to submit on #359 and #361 so I think this one (which is really about the white screen) can be closed as I can't recreate it. Does that seem reasonable?  
#359 and #361 are needed before we can deploy so if you can/are working those can you please bring them over to in progress?

**Comment by calebissharp** [comment]
@LisaMoore1 This one should be alright to close.
